### PR TITLE
MULE-10401: Remove HTTP parsing mechanism.

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

File: tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java [hunk 1]
```diff
@@ -88,6 +91,13 @@ public void init() {
         .build());
 
     componentBuildingDefinitions.add(baseDefinition.copy()
+        .withIdentifier("exception-strategy")
```
Comment (elrodro83) [inline tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java]: why readd this?

File: tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java [hunk 2]
```diff
@@ -149,6 +159,12 @@ public void init() {
         .withTypeDefinition(fromType(SharedConfig.class))
         .build());
 
+    componentBuildingDefinitions
```
Comment (elrodro83) [inline tests/component-plugin/src/main/java/org/mule/functional/config/TestComponentBuildingDefinitionProvider.java]: why this change?

File: tests/component-plugin/src/main/resources/META-INF/mule-test.xsd [hunk 3]
```diff
@@ -301,4 +301,40 @@
         </xsd:complexContent>
     </xsd:complexType>
 
+    <xsd:element name="set-payload" type="setPayloadTransformerType" substitutionGroup="mule:abstract-message-processor">
```
Comment (elrodro83) [inline tests/component-plugin/src/main/resources/META-INF/mule-test.xsd]: why this?

Comment (afelisatti) [inline tests/component-plugin/src/main/resources/META-INF/mule-test.xsd]: Since `set-payload` is going to become unavailable here soon and `transform` won't be available in CE I thought we should have a test version to make tests (and their migration) easier. When the `set-payload` task is done the Processor will just need to be moved to the test plugin.
